### PR TITLE
Add edge case feeding tests and refactor coordinator

### PR DESCRIPTION
**Pull request by Bigdaddy1990**
## Summary
- test FeedingManager with `None` meal types and prior-day feedings
- refactor coordinator to calculate daily feedings once

## Testing
- `pip install -r requirements_test.txt`
- `pre-commit run --files tests/test_feeding_manager.py custom_components/pawcontrol/coordinator.py`
- `pytest --no-cov -q`

------
https://chatgpt.com/codex/tasks/task_e_68b4f723f69483318ddeb563b961e54c

### CODE REVIEW

File: tests/test_feeding_manager.py [hunk 1]
```diff
@@ -46,3 +48,34 @@ async def test_feeding_manager_empty_history() -> None:
...
+
+    # Feeding yesterday should be ignored
+    await manager.async_add_feeding("dog", 1.0, meal_type="breakfast", time=yesterday)
+    # Feeding today should be counted
+    await manager.async_add_feeding("dog", 1.0, meal_type="dinner")
+
+    data = await manager.async_get_feeding_data("dog")
+
+    assert data["feedings_today"] == {"dinner": 1}
+    assert data["total_feedings_today"] == 1
```
[the remark this answers is not in the document]
Comment (Bigdaddy1990) [reply tests/test_feeding_manager.py]: @codex
